Setup add widgets advanced guide (#1283)
CartoDB/carto-vl: docs/guides/12-add-widgets-advanced.md

## Add widgets advanced

We have seen how to add legends to our map by using `getLegendsData` method, and we have also learned how to display widgets using histogram expressions in the [previous guide](/developers/carto-vl/guides/add-widgets/). This is an advanced guide to learn how to obtain and display the information in widgets by using `viewportHistogram` and `globalHistogram` expressions.

### Overview

> We will learn how to create bar charts for **categorical** data and histograms for **numeric** data. For this purpose, the following examples make use of the same source, the [**Vancouver Trees**](https://team.carto.com/u/cartovl/tables/cartovl.vancouver_trees/public/map) dataset.

Reviewer: Nice dataset. But I have seen that it has some wrong features, in our beloved [Null Island](https://en.wikipedia.org/wiki/Null_Island) at (0, 0), so we should probably delete those.

> Bar charts and histograms have been built with [Chart.js](https://www.chartjs.org), a very simple external library. In the previous guide we used [Airship](https://carto.com/airship/) to build a histogram widget. In this case, we want to demonstrate you can use the visualization library of your choice with CARTO VL, and therefore, we have to explain how the histogram expressions work a bit deeply.

Reviewer: ```suggestion
Bar charts and histograms have been built with [Chart.js](https://www.chartjs.org), a very simple external library. In the previous guide we used [Airship](https://carto.com/airship/), which is the natural choice to build a histogram widget, but in this case we want to demonstrate how the histogram expressions work a bit deeply.
```

I would stress a bit the idea of VL <--> Airship here. That will be improved a lot with the Airship v2 VL-Bridge

Author: I agree, but I think this is not the place to do so, or maybe not at this time. Maybe we can modify this when there's a VL- Airship bridges guide, what do yo think? Explaining something that is not yet available adds a bit of noise, in my opinion.

### Histogram expressions

> There are two histogram expressions in CARTO VL: `viewportHistogram` and `globalHistogram`. These expressions return a list of values grouped by column. The `viewportHistogram` returns the list based on the features that are visible in the viewport, while `globalHistogram` takes into account the whole dataset.

Reviewer: Regarding to 'whole dataset', I think we could add here a note on the stats and how the histogram is calculated

Author: Agree! Good place to explain this.

      }
    }],
    xAxes: [{
      gridLines: {
        drawBorder: false,
        display: false
      },
      ticks: {
        display: false
      },
      barPercentage: 1.0,
      categoryPercentage: 1.0
    }]
  }
};

const chart = new Chart(ctx, {
  type: 'bar',
  options: chartOptionsDefault
});
```

The `histogram.value` returns an array of `{ x, y }` objects where `x` is the name of the category and `y` is the amount of trees for this category. We are going to build `data` and `labels` arrays from the histogram expression  **once the layer is updated**:

```js
layer.on('updated', () => {
  // Get histogram data
  const histogramData = layer.viz.variables.v_histogram.value;

  // Chart.js set up
  const labels = histogramData.map(elem => elem.x);
  const data = histogramData.map(elem => elem.y);
  const backgroundColor = ['#00718b', '#089099', '#46aea0', '#7ccba2', '#f7feae'];

  chart.data = {
    labels,
    datasets: [{
      data,
      backgroundColor
    }]
  };

  chart.update();
});
```

Here you can see the result. Check how the bars change when you interact with the map.

<div class="example-map">
  <iframe
    id="guides-widgets-advanced-step-1"

>     src="/developers/carto-vl/examples/maps/guides/add-widgets-advanced/step-1.html"

Reviewer: In this first example all the bars have the same teal color, so I guess that the previous code doesn't correspond to thist first step.
